app/backend/hiStoryBackend/api/views/auth/views_v1.py

```diff
@@ -1,16 +1,49 @@
 from django.http import JsonResponse
 from api.helpers import json_response_helper as jrh
-
+from api.helpers import custom_helper
+from django.contrib.auth.hashers import check_password, make_password
+import json
+import jwt
+from api.models import User
+from django.core.exceptions import ValidationError
 
 def sign_in(request):
     if not request.method == 'POST':
         return jrh.bad_request(['Unsupported HTTP verb'])
 
     return JsonResponse({'operation': 'sign-in'})
 
-
 def sign_up(request):
     if not request.method == 'POST':
         return jrh.bad_request(['Unsupported HTTP verb'])
 
-    return JsonResponse({'operation': 'sign-up'})
+    r = json.loads(request.body)
+
+    username = r.get('username')
+    email = r.get('email')
+    password = r.get('password')
+    password_confirmation = r.get('password_confirmation')
+    full_name = r.get('full_name')
+
+    if(password != password_confirmation):
+    	return jrh.fail(["Passwords don't match."])
+
+    password_hash = make_password(password)
```

Comment:
I just realized that when both `password` and `password_confirmation` keys are missing from the request, they will be set as `None` (hence, they are consired to be equal). When `make_password()` is run with the parameter `None`, it generates a random hash.
Also, current constraints at the `model.py` don't handle those requirements:
> 1.3.6.1 Passwords shall contain at least 8 characters.
> 1.3.6.2 Passwords shall contain at most 24 characters.

Can you implement this logic on here for now?

Reply:
One quick question. Does one need to check if the username field exists in the incoming json?

Reply:
> One quick question. Does one need to check if the username field exists in the incoming json?

Because I did do checks, if they are not required, I won't do so in the future

Reply:
I will implement the changes asap. When one sends a request without username or email, u.full_clean() throws an error saying that they shouldn't be null.

Reply:
> One quick question. Does one need to check if the username field exists in the incoming json?

Server should handle the request properly (i.e. without throwing an error/exception) when an expected key is missing in the request.